obspy/core/event/base.py

```diff
@@ -611,43 +667,73 @@ def get_referred_object(self):
         Will return None if no object could be found.
         """
```

Comment:
I might be overthinking or over-complicating things here, but some of logic breaks down in a threaded context and there are multiple places where race conditions could happen.

For the resource ids that might actually happen fairly often as some users download in threads (I know I do). It might be worthwhile to add a class level `Threading.RLock` and wrap some methods in

```python
def get_referred_object(...):
    with self.__class__.__lock:
        ...
```

or maybe even a decorator? I think wrapping the `__init__()`, `get_referred_object()`, and `set_referred_object()` methods would take care of most race conditions.